src/main/java/de/tum/in/www1/artemis/service/DataExportService.java

```diff
@@ -0,0 +1,567 @@
+        this.postRepository = postRepository;
+        this.answerPostRepository = answerPostRepository;
+        this.reactionRepository = reactionRepository;
+    }
+
+    /**
+     * Requests a data export for the given user.
+     * This will create a new DataExport object in the database and start the creation of the data export.
+     *
+     * @param user the user for which to create the data export
+     * @return the created DataExport object
+     **/
+    public DataExport requestDataExport(User user) throws IOException {
+        DataExport dataExport = new DataExport();
+        dataExport.setDataExportState(DataExportState.REQUESTED);
+        dataExport.setUser(user);
+        dataExport.setRequestDate(ZonedDateTime.now());
+        if (!Files.exists(dataExportPath)) {
+            Files.createDirectories(dataExportPath);
+        }
+        dataExportRepository.save(dataExport);
+        workingDirectory = Files.createTempDirectory(dataExportPath, "data-export-working-dir");
+        dataExport.setDataExportState(DataExportState.IN_CREATION);
+        dataExportRepository.save(dataExport);
+        var dataExportPath = createDataExport(user);
+        dataExport.setFilePath(dataExportPath.toString());
+        // sending the email will be part of a follow-up, for now this just implies export finished
+        dataExport.setCreationDate(ZonedDateTime.now());
+        dataExport.setDataExportState(DataExportState.EMAIL_SENT);
+        dataExport = dataExportRepository.save(dataExport);
+        return dataExport;
+    }
+
+    /**
+     * Download the data export for the given user id and data export id.
+     *
+     * @param userId       the id of the user for which to download the data export
+     * @param dataExportId the id of the data export to download
+     * @return the file path where the data export is stored
+     * @throws de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException  if the data export or the user could not be found
+     * @throws de.tum.in.www1.artemis.web.rest.errors.AccessForbiddenException if the user is not allowed to download the data export
+     */
+    public Path downloadDataExport(long userId, long dataExportId) {
+        var dataExport = dataExportRepository.findByIdElseThrow(dataExportId);
+        var user = userRepository.findByIdElseThrow(userId);
+        // check data export belongs to specified user
+        authorizationCheckService.isOwnerOfDataExportElseThrow(dataExport, user);
+        // check data export belongs to currently logged-in user
+        authorizationCheckService.currentlyLoggedInUserIsOwnerOfDataExportElseThrow(dataExport);
+
+        // the first condition checks that the export has been created and the second that it has not yet been deleted (we allow multiple downloads as long as the export is not
+        // deleted
+        if (dataExport.getDataExportState() != DataExportState.EMAIL_SENT && dataExport.getDataExportState() != DataExportState.DOWNLOADED) {
```

Comment:
Maybe move this into the DataExportState as a method isInDownloadableState or something like that